### ISSUE TITLE
Add cmake interface

### ISSUE THREAD

**Issue by jbmouret** [resibots/limbo#313]
We did this in robot_dart and it makes it easier to compile with cmake projects.

**Comment by mmattamala** [comment]
Hi @jbmouret 

Thanks for sharing the library. It was on my radar a while ago and now I found some applications to explore its use.
I've been investigating building the library with CMake, since my plan was to use catkin to build it. 

So far I'm using CMake's `ExternalProject` to run `waf` and then setting up the static binary and headers using CMake. Then a `package.xml` file handles integration with catkin (I was pointed to the [REP-136](https://www.ros.org/reps/rep-0136.html) to do this). 
I'm working on a [separate fork](https://github.com/mmattamala/limbo), but it's still quite messy because I tried many things, including the waftools Cmake stuff mentioned in https://github.com/resibots/limbo/issues/261

By doing the this I was able to link limbo and interpolate using GPs with fixed kernels. However, now I'm struggling to get the NLOpt support to train the hyperparameters, since I get the following warning when trying to build the package that includes limbo:
```bash
warning: #warning NO NLOpt, and NO Libcmaes: the acquisition function will be optimized by a grid search algorithm (which is usually bad). Please install at least NLOpt or libcmaes to use limbo!
```

Do you have any suggestion on something that I'm probably missing that you internally handle with waf?
I'm no CMake expert (I usually setup my files using other as a reference) so any comment on that could be helpful.

Thanks!

**Comment by jbmouret** [comment]
Hi

First, most of limbo is header-only, so you could simply include the files. Honestly, this is what I would do... I should try to find some time to write some documentation about this.

Indeed, you will need to compile limbo with NLOpt and/or CMA-Es for BO. To do so, you need to:
- install them [see here http://www.resibots.eu/limbo/tutorials/compilation.html ]
- if you are compiling limbo with waf, it should detect NLOpt and CMA-ES automatically during the configure
- compile limbo **and your code** with -DUSE_NLOPT -DUSE_LIBCMAES (this is because most of limbo is in the headers)

**Comment by mmattamala** [comment]
Hi, thanks for the reply, I was missing the `-DUSE_NLOPT` flag in the installation config files. 
Now I'm just using the headers in CMake and I can easily import it with catkin in other projects. Cheers!

**Comment by IoannisDadiotis** [comment]
Hi all,

I just installed limbo and I would like to use it within my catkin based ros projects.

@jbmouret is there any documentation now on this?

@mmattamala is there any example publicly available of how you interfaced limbo with your cmake/catkin-based project? 

Thanks in advance

**Comment by mmattamala** [comment]
Hi @IoannisDadiotis , I have a fork in my account: https://github.com/mmattamala/limbo
It should compile by `catkin build limbo`. Let me know if you have further questions.

**Comment by costashatz** [comment]
@mmattamala this is very cool. Would you consider making a PR so that the rest of the users can profit from this one? Thanks!

**Comment by mmattamala** [comment]
Yes, for sure. I'll do it now. Thank you!

**Comment by IoannisDadiotis** [comment]
> Hi @IoannisDadiotis , I have a fork in my account: https://github.com/mmattamala/limbo It should compile by `catkin build limbo`. Let me know if you have further questions.

Thanks @mmattamala .

Do you think just `catkin build limbo` is enough or I still have to do ` ./waf configure` and `./waf build` before?

**Comment by costashatz** [comment]
This is fixed by #330 !
